libmogwai-schedule/schedule-service.c

```diff
@@ -452,17 +457,29 @@ count_entries (MwsScheduleService *self,
 }
 
 static void
-notify_scheduler_properties (MwsScheduleService *self)
+notify_scheduler_properties (MwsScheduleService *self,
+                             gboolean            notify_entries,
+                             gboolean            notify_allow_downloads)
 {
   g_auto(GVariantDict) changed_properties_dict = G_VARIANT_DICT_INIT (NULL);
   guint32 entries, active_entries;
 
-  count_entries (self, &entries, &active_entries);
+  if (notify_entries)
+    {
+      count_entries (self, &entries, &active_entries);
 
-  g_variant_dict_insert (&changed_properties_dict,
-                         "ActiveEntryCount", "u", active_entries);
-  g_variant_dict_insert (&changed_properties_dict,
-                         "EntryCount", "u", entries);
+      g_variant_dict_insert (&changed_properties_dict,
+                             "ActiveEntryCount", "u", active_entries);
+      g_variant_dict_insert (&changed_properties_dict,
+                             "EntryCount", "u", entries);
+    }
+
+  if (notify_allow_downloads)
+    {
+      g_variant_dict_insert (&changed_properties_dict,
+                             "DownloadsAllowed", "b",
```

Comment:
I don't see what we gain from changing the form here.

Reply:
You mean changing the name from `allow-downloads` to `DownloadsAllowed`? My thinking was that `AllowDownloads` sounds like a property you could change the value of, rather than a read-only one. `DownloadsAllowed` sounds more read-only.

Reply:
Yeah, I meant calling it `AllowDownloads` for consistency (as, in any case, usually I always have to read whether properties are RO or RW...). In any case, not a strong opinion as I said.